docs/code_annotations.rst

```diff
@@ -0,0 +1,196 @@
+Code Annotations
+================
+
+A special directive was created to make use of the mkdocs-material theme's
+`Code Annotations <https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations>`_
+feature.
+
+.. rst:directive:: code-annotations
+
+    This directive shall hold an :duref:`enumerated list <enumerated-lists>` of annotations that can be used in a
+    `code-block` immediately prior to this directive. The given list will not be rendered in the
+    generated output, unless there is no immediately prior `code-block` to annotate or there are no
+    annotated comments in the immediately prior `code-block`. Additionally, all annotations will be
+    excluded when using the "copy to clipboard" button for an annotated `code-block`; the annotated
+    comments' text will still be copied (if not `Hiding the annotated comment`_).
+
+    An error will be shown in the build log if the given list is not recognized as an
+    :duref:`enumerated list <enumerated-lists>`.
+
+    .. note::
+        When using this directive, the :python:`"content.code.annotate"` does not need to be
+        specified in the :themeconf:`features` list.
+
+Using annotations
+*****************
+
+Each annotation in the code snippet should be a comment containing the identifying number in
+parentheses (``(1)``) that corresponds to the sequential order of annotations specified in the
+list of :rst:dir:`code-annotations`.
+
+.. note::
+    Remember to specify the language syntax for pygments to highlight the `code-block`. Without
+    this information, pygments may incorrectly identify comments in a `code-block` which is
+    required for the detection of annotations.
+
+.. hint::
+    All annotations are hyperlinked. This means you can right click the annotation's button and
+    open it as a link in a new tab (or copy the link to share it).
+
+.. rst-example:: Example of code annotations
+
+    .. code-block:: python
+
+        html_theme_options = {
+            "features" : [
+                "content.code.annotate"  # (1)
+            ],
+        }
+
+    .. code-annotations::
+        1. .. admonition:: Obsolete
+               :class: failure
+
+               This has no special effect because the :rst:dir:`code-annotations` directive
+               automatically enables the feature.
+
+Arbitrary list markers
+----------------------
+
+The :duref:`enumerated lists' <enumerated-lists>` markers are arbitrary and do not directly
+correspond to the identifying numbers used in the annotated comment. Because of this arbitration,
```

Comment:
arbitration means something else --- you could use "arbitrariness" instead, or perhaps a different phrasing.

Also in next paragraph.